pkg/framework/deployment.go

```diff
@@ -43,6 +44,17 @@ func DeleteDeployment(c client.Client, deployment *kappsapi.Deployment) error {
 	})
 }
 
+// UpdateDeployment updates the specified deployment
+func UpdateDeployment(c client.Client, deployment *kappsapi.Deployment, updated *kappsapi.Deployment) error {
+	return wait.PollImmediate(1*time.Second, WaitMedium, func() (bool, error) {
+		if err := c.Patch(context.TODO(), deployment.DeepCopy(), client.MergeFrom(updated)); err != nil {
+			glog.Errorf("error querying api for deployment object %q: %v, retrying...", deployment.Name, err)
```

Comment:
We aren't querying here, we are patching, perhaps this is better?
```suggestion
			klog.Errorf("error patching deployment object %q: %v, retrying...", deployment.Name, err)
```